sdk/src/main/java/com/bugsnag/android/ErrorStore.java

```diff
@@ -133,22 +145,19 @@ void flushReports(Collection<File> storedReports) {
 
     private void flushErrorReport(File errorFile) {
         try {
-            Report report = null;
-            if (config.getBeforeSendTasks().size() > 0) {
```

Comment:
You mentioned in the PR description that "files are now always deserialised" but I don't know why this change was made… could you elaborate?

Reply:
Previously there were two codepaths here:

1. If zero `beforeSendCallbacks` were set on the notifier, then we'd stream the error directly from disk to a network connection.
2. If `beforeSendCallbacks` are set on the notifier we'd deserialise the file into an `Error` object, then serialise that as JSON when making the request.

If the file has been corrupted then deserialisation of JSON should fail - at which point we need to send a minimal error report.